packages/firebase_vertexai/firebase_vertexai/lib/src/api.dart

```diff
@@ -607,6 +616,40 @@ enum HarmBlockThreshold {
   Object toJson() => _jsonString;
 }
 
+/// Specifies how the block method computes the score that will be compared
+/// against the [HarmBlockThreshold] in [SafetySetting].
+enum HarmBlockMethod {
+  /// The harm block method uses both probability and severity scores.
+  severity('SEVERITY'),
+
+  /// The harm block method uses the probability score.
+  probability('PROBABILITY'),
+
+  /// The harm block method is unspecified.
+  unspecified('HARM_BLOCK_METHOD_UNSPECIFIED');
+
+  const HarmBlockMethod(this._jsonString);
+
+  // ignore: unused_element
+  static HarmBlockMethod _parseValue(Object jsonObject) {
+    return switch (jsonObject) {
+      'SEVERITY' => HarmBlockMethod.severity,
+      'PROBABILITY' => HarmBlockMethod.probability,
+      'HARM_BLOCK_METHOD_UNSPECIFIED' => HarmBlockMethod.unspecified,
+      _ =>
+        throw FormatException('Unhandled HarmBlockMethod format', jsonObject),
+    };
+  }
+
+  final String _jsonString;
+
+  @override
+  String toString() => name;
+
+  /// Convert to json format.
```

Comment:
[nit] This comment isn't adding new information beyond the name. Consider removing it if there isn't any futher information worth adding.